packages/indexer-agent/src/network.ts

```diff
@@ -714,16 +713,27 @@ export class Network {
                           deploymentRule.minAverageQueryFees,
                         ).toString()
                       : null,
+                    requireSupported: deploymentRule.requireSupported,
                   },
                 })
 
                 // Skip the indexing rules checks if the decision basis is 'always' or 'never'
                 if (
-                  deploymentRule.decisionBasis === IndexingDecisionBasis.ALWAYS
+                  deploymentRule?.decisionBasis === IndexingDecisionBasis.ALWAYS
                 ) {
+                  // cannot allocate to unsupported subgraph by default
+                  if (
```

Comment:
I think this check should be moved outside of the `deploymentRule?.decisionBasis === IndexingDecisionBasis.ALWAYS` conditional block, so it stops all types of rules from allocating towards `unsupported` subgraphs.